packages/opentelemetry-resources/src/platform/node/detectors/EnvDetector.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Resource, Labels } from '../../../Resource';
+
+/**
+ * EnvDetector can be used to detect the presence of and create a Resource
+ * from the OTEL_RESOURCE_LABELS environment variable.
+ */
+export class EnvDetector {
+  // Type, label keys, and label values should not exceed 256 characters.
+  private static readonly _MAX_LENGTH = 255;
+
+  // OTEL_RESOURCE_LABELS is a comma-separated list of labels.
+  private static readonly _COMMA_SEPARATOR = ',';
+
+  // OTEL_RESOURCE_LABELS contains key value pair separated by '='.
+  private static readonly _LABEL_KEY_VALUE_SPLITTER = '=';
+
+  private static readonly _ERROR_MESSAGE_INVALID_CHARS =
+    'should be a ASCII string with a length greater than 0 and not exceed ' +
+    EnvDetector._MAX_LENGTH +
+    ' characters.';
+
+  private static readonly _ERROR_MESSAGE_INVALID_VALUE =
+    'should be a ASCII string with a length not exceed ' +
+    EnvDetector._MAX_LENGTH +
+    ' characters.';
+
+  /**
+   * Returns a {@link Resource} populated with labels from the
+   * OTEL_RESOURCE_LABELS environment variable. Note this is an async function
+   * to conform to the Detector interface.
```

Comment:
You mention a Detector interface, but none of the detectors implements an interface.